codec/legacy/codec.go

```diff
@@ -10,10 +10,9 @@ import (
 // has all Tendermint crypto and evidence types registered.
 //
 // TODO: Deprecated - remove this global.
-var Cdc *codec.LegacyAmino
+var Cdc = codec.NewLegacyAmino()
```

Comment:
This is just to make sure the Cdc is initialized before running the various `init` methods. Since the initialization order is the following:

1. package variables (as declared) 
2. `init` methods (based on the file structure)

It is safer to have the Cdc being initialized as a package variable rather than inside the `init` method.

Reply:
Initialization order is not necessarily defined in this way.

Reply:
Isn't it? I found this order inside the Go specs ([here](https://go.dev/ref/spec#Package_initialization))

Reply:
You're right that vars are evaluated before init funcs, my mistake there. But this is very subtle behavior to depend on.

Reply:
This honestly wasn't necessary. I actually think keeping it in the `init` is cleaner as that's also where we do the initialization.